model/PciRegistry.php

```diff
@@ -50,267 +44,378 @@ class PciRegistry extends ConfigurableService
     }
-    
+
     /**
-     * 
-     * @todo
-     * @param string $typeIdentifier
-     * @param string $sourceVersion
-     * @param string $targetVersion
-     * @param array $runtime
-     * @param array $creator
+     *
+     * @param PciModel $pciModel
+     * @return array
+     * @throws \common_Exception
      */
-    public function update($typeIdentifier, $sourceVersion, $targetVersion, $runtime = [], $creator = []){
-        
+    protected function getFilesFromPci(PciModel $pciModel)
+    {
+        $validator = new PciModelValidator($pciModel);
+        return $validator->getRequiredAssets();
     }
-    
-    public function getBaseUrl($typeIdentifier, $version = ''){
-        if($this->exists($typeIdentifier, $version)){
+
+    /**
+     * Return the absolute url of PCI storage
+     *
+     * @param $typeIdentifier
+     * @param string $version
+     * @return bool|string
+     */
+    protected function getBaseUrl($typeIdentifier, $version = null)
+    {
+        $pciModel = new PciModel($typeIdentifier, $version);
+        if ($this->exists($pciModel)) {
             return $this->getFileUrl($typeIdentifier, $version, '');
         }
         return false;
     }
-    
+
     /**
-     * Unregsiter an single pci. 
-     * If the version is '*', all pci version will be unregistered
-     * 
-     * @todo
+     * Unregister PCI by removing the given version data & asset files
+     * If $pciModel doesn't have version, all versions will be removed
+     *
+     * @param PciModel $pciModel
+     * @return bool
+     * @throws \common_Exception
      */
-    public function unregister($typeIdentifier, $version = '*'){
-        
+    public function unregister(PciModel $pciModel)
+    {
+        if (!$this->exists($pciModel)) {
+            throw new \InvalidArgumentException('Identifier "' . $pciModel->getTypeIdentifier() . '" to remove is not found in PCI map');
+        }
+
+        $this->removeAssets($pciModel);
+        $this->removeMapPci($pciModel);
+        return true;
     }
-    
+
     /**
-     * Return all file contents, following the same array format as the method getRuntime()
-     * This method is useful to export registered PCI into standard QTI item packages
-     * 
-     * @todo
-     * @param type $typeIdentifier
-     * @param type $version
+     * Return the runtime of PCI
+     *
+     * @param $typeIdentifier
+     * @param string $version
+     * @return array|string
+     * @throws \common_Exception
      */
-    public function export($typeIdentifier, $version){
-        
-    }
-    
-    public function getRuntime($typeIdentifier, $version = ''){
+    protected function getRuntime($typeIdentifier, $version = '')
+    {
         $pcis = $this->getMap();
-        if(isset($pcis[$typeIdentifier])){
-            if(empty($version)){
+        if (isset($pcis[$typeIdentifier])) {
+            if (empty($version)) {
                 $version = $this->getLatestVersion($typeIdentifier);
```

Comment:
$version = $this->getLatestVersion($typeIdentifier)->getVersion();
Since we have many concept now, maybe we should follow a consistent naming convetion : $version used to be a string following the semantic versionning format x.y.z but now it is more an instance of pciModel which is quite confusing...